spec/integration/user_index_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+RSpec.describe 'User index integration tests', type: :system do
+  before do
+    driven_by(:rack_test)
+  end
+  before(:all) do
+    @users = User.all
+  end
+  it 'shows all the properties (Name, photo, number of posts) of each user' do
+    visit users_path
+    @users.each do |user|
+      expect(page).to have_text(user.Name)
+      expect(page).to have_content('Number of posts')
+
+      expect(page).to have_text(user.PostsCounter)
+    end
+  end
+  it 'shows the number  of  posts for the user' do
+    visit '/users'
+    expect(page).to have_content('Number of posts')
+  end
```

Comment:
- [ ] Great job adding some of the tests for user index page as per project requirements. You are missing the the `expect` statement to check if a user has a photo. Kindly add it.
- [ ] You are supposed to add a test that redirects to the user's show page when you click on a user. This test is missing for your user index page. Please kindly add it.